db-scheduler-ui/src/main/java/com/github/bekk/dbscheduleruiapi/service/LogLogic.java

```diff
@@ -17,27 +19,63 @@
 public class LogLogic {
 
   private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
-  private final JdbcTemplate jdbcTemplate;
 
   @Autowired
   public LogLogic(DataSource dataSource) {
     this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
-    this.jdbcTemplate = new JdbcTemplate(dataSource);
   }
 
-  public List<LogModel> getLogsById(String taskName, String taskInstance) {
+  public List<LogModel> getLogsById(TaskDetailsRequestParams requestParams) {
     Map<String, Object> params = new HashMap<>();
-    params.put("taskName", taskName);
-    params.put("taskInstance", taskInstance);
-    return namedParameterJdbcTemplate.query(
-        "SELECT * FROM scheduled_execution_logs WHERE task_name = :taskName AND task_instance = :taskInstance ORDER BY time_started DESC",
-        params,
-        new LogModelRowMapper());
+    params.put("taskName", requestParams.getTaskName());
+    params.put("taskInstance", requestParams.getTaskId());
+    StringBuilder baseQuery =
+        new StringBuilder(
+            "SELECT * FROM scheduled_execution_logs WHERE task_name = :taskName AND task_instance = :taskInstance");
+
+    if (requestParams.getSearchTerm() != null && !requestParams.getSearchTerm().trim().isEmpty()) {
+      baseQuery.append(
+          " AND (LOWER(id) LIKE LOWER(:searchTerm) OR LOWER(task_name) LIKE LOWER(:searchTerm) OR LOWER(task_instance) LIKE LOWER(:searchTerm))");
```

Comment:
Skal vi søke på execution id? Bruker vi den noe sted?
![image](https://github.com/bekk/db-scheduler-ui/assets/136692676/916aad5d-76c5-4e8d-8e72-f36fa08c575a)

Reply:
Vet ikke om den er tilgjengelig noe sted utenom ui'et vårt, så måtte isåfall være at man lagrer den for å søke på den senere